api/V1/eventUtil.ts

```diff
@@ -79,8 +112,80 @@ const eventAuth = [
   )
 ];
 
+export class PowerEvents {
+  lastReported: Moment | null;
+  lastStarted: Moment | null;
+  lastStopped: Moment | null;
+  Device: Device | null;
+  hasStopped: boolean;
+  constructor(event: Event) {
+    this.hasStopped = false;
+    this.lastReported = null;
+    this.lastStarted = null;
+    this.lastStopped = null;
+    this.update(event);
+  }
+  update(event: Event) {
+    if (!this.Device && event.Device) {
+      this.Device = event.Device;
+    }
+    const eventDate = moment(event.dateTime);
+    switch (event.EventDetail.type) {
+      case "rpi-power-on":
+        if (this.lastStarted == null || eventDate.isAfter(this.lastStarted)) {
+          this.lastStarted = eventDate;
+        }
+        break;
+      case "daytime-power-off":
+        if (this.lastStopped == null || eventDate.isAfter(this.lastStopped)) {
+          this.lastStopped = eventDate;
+        }
+        break;
+      case "stop-reported":
+        if (this.lastReported == null || eventDate.isAfter(this.lastReported)) {
+          this.lastReported = eventDate;
+        }
+    }
+  }
+  checkIfStopped(): boolean {
+    if (this.lastStarted == null) {
+      this.hasStopped = false;
+      return this.hasStopped;
+    }
+    let hasStopped = false;
+
+    // check that started hasn't occured after stopped
+    if (
```

Comment:
Decouple this if statement, 

It should be:
if (laststopped == null) 
..
else if this.lastStarted.isAfter....

else

It makes no logical sense to combine the first two if statements then separate them.